src/dfind/dfind.c

```diff
@@ -542,25 +562,89 @@ int main (int argc, char** argv)
     	}
     }
 
+    /* print usage if we need to */
+    if (usage) {
+        if (rank == 0) {
+            print_usage();
+        }
+        mfu_file_delete(&mfu_file);
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    char** argpaths = &argv[optind];
+
+    /* The remaining arguments are treated as paths */
+    int numpaths = argc - optind;
+
+    /* advance to next set of options */
+    optind += numpaths;
+
+    /* Before processing, make sure we have at least one path to begin with */
+    if (numpaths < 1) {
+        if (rank == 0) {
+            MFU_LOG(MFU_LOG_ERR, "At least one path is needed");
+            print_usage();
+        }
+        mfu_file_delete(&mfu_file);
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
```

Comment:
Some tools can work from just an input file -- the input file replaces the need to walk a path.  I think dfind should be one of those, e.g.,
```
dwalk --output list.mfu /path/to/walk
dfind --input list.mfu --output filesonly.mfu --type f
```
I haven't checked this logic in detail, but would this require a path?

Reply:
You're right. This check was preventing `--input` from being used with `--output`. I instead moved a check into `daos_setup` to make sure we have enough paths there. Otherwise, the path checking falls into the pre-existing logic for each tool.
Thanks for catching this!